src/librbd/PluginRegistry.cc

```diff
@@ -45,14 +45,54 @@ void PluginRegistry<I>::init(const std::string& plugins, Context* on_finish) {
       break;
     }
 
-    m_plugin_hook_points.emplace_back();
-    auto hook_points = &m_plugin_hook_points.back();
-    plugin->init(m_image_ctx, *m_plugin_api, hook_points, ctx);
+    plugin->init(m_image_ctx, *m_plugin_api, m_plugin_hook_points, ctx);
   }
 
   gather_ctx->activate();
 }
 
+template <typename I>
+void PluginRegistry<I>::open_hook_points(Context* on_finish) {
```

Comment:
Nit: rename to make the actual hook-point here and below (e.g. `acquire_exclusive_lock` and `release_exclusive_lock`). We might want to event prefix both of the hook points (and these methods) w/ `handle_` since the hook is reacting to the exclusve-lock vs actually attempting to acquire/release it.

Reply:
How about send_post_exclusive_hook_points() and handle_send_post_exclusive_hook_points()?

Reply:
Renamed.